pkg/registry/generic/etcd/etcd.go

```diff
@@ -259,7 +259,17 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 	// TODO: expose TTL
 	creating := false
 	out := e.NewFunc()
-	err = e.Storage.GuaranteedUpdate(ctx, key, out, true, func(existing runtime.Object, res storage.ResponseMeta) (runtime.Object, *uint64, error) {
+	meta, err := api.ObjectMetaFor(obj)
+	if err != nil {
+		return nil, false, kubeerr.NewInternalError(err)
+	}
+	var preconditions *storage.Preconditions
+	// If the UID of the new object is specified, we use it as an Update precondition.
+	if len(meta.UID) != 0 {
```

Comment:
It seems we cannot force user to support an UID (https://github.com/kubernetes/kubernetes/issues/23297#issuecomment-199506793), so we only use it as a precondition if it's not empty.